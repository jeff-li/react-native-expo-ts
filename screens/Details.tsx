import React from 'react';
import { StyleSheet, View, Text, Image } from 'react-native';
import { Avatar } from "native-base"
import { useRoute, RouteProp } from '@react-navigation/native';
import { HomeStackParamList } from '../types';

type DetailsRouteProp = RouteProp<
  HomeStackParamList,
  'Details'
>;

export default function Details() {
  const route = useRoute<DetailsRouteProp>();
  const data = route.params.data || {};
  return (
    <View style={styles.container}>
      <View style={styles.image}>
        <Avatar
          size="100px"
          source={{
            uri: data.avatarUrl,
          }}
        />
      </View>
      <Text style={styles.text}>{data.fullName}</Text>
      <Text style={styles.text}>{data.recentText}</Text>
      <Text style={styles.text}>{data.timeStamp}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ebebeb'
  },
  text: {
    color: '#333',
    fontSize: 24,
    fontWeight: 'bold'
  },
  image: {
    paddingBottom: 50
  }
});