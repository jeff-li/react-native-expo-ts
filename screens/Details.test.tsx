import React from "react";
import { render } from '@testing-library/react-native';

import Details from "./Details";

describe("<Details />", () => {
  it("should match screenshot", () => {
    const { toJSON, getAllByText } = render(
      <Details />
    );
    expect(toJSON()).toMatchSnapshot()
    const bananaElements = getAllByText('Details Screen');
    expect(bananaElements).toHaveLength(1);
  });
});