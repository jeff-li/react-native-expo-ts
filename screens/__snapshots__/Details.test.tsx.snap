// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should match screenshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ebebeb",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Details Screen
  </Text>
</View>
`;
